game/state/tilemap/tilemap.h

```diff
@@ -76,6 +77,8 @@ class TileMap
   private:
 	std::vector<Tile> tiles;
 	std::vector<std::set<TileObject::Type>> layerMap;
+	// Cache for the strategy view mode.
+	std::vector<sp<Surface>> strategyViewCache;
```

Comment:
I'm not sure if I like the cached images being stored in the game/state - it feels very much like a UI-specific thing. Perhaps this is the best place to keep the "dirty" state though.

Reply:
I understand your point.
Possible approach may be to create cache in the TileView. But in this case the TileObjectScenery and the TileObjectBattleMapPart should to know about TileView (or at least about the cache). It's not so easy to find a clever way to do this.

The TileMap lies in the key place which can be easily accessed as for CityTileView and for the BattleTileView and for TileObjects too. Thus the cache easely accessed for each who should to use it.